server/logging.go

```diff
@@ -0,0 +1,93 @@
+package server
+
+import (
+	"fmt"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	"strings"
+)
+
+func NewMlogLogger(cfg *ConfigPushProxy) (*mlog.Logger, error) {
+	// Initialize the logger - begin
+	logger, err := mlog.NewLogger()
+	if err != nil {
+		return nil, err
+	}
+	cfgJSON := cfg.LoggingCfgJSON
+	if cfg.LoggingCfgFile == "" && cfgJSON == "" {
+		// if no logging defined, use default config (console output)
+		cfgJSON = defaultLoggingConfig(cfg)
+	}
```

Comment:
There is no need to have a long JSON file and then pass that json string to `logger.Configure`. A better way is to use `logger.ConfigureTargets` after specifying the targets for file and console.

And to make each target, you can simply use `mlog.TargetCfg` to have better type safety.

So create logger with `mlog.NewLogger()`. Then create a logger config map of `mlog.LoggerConfiguration`, and add two targets. One for console and one for file.

For example:

```go
mlog.TargetCfg{
  Type:         "console",// console or file
  Levels:       mlog.StdAll,
  Format: "json", // json or plain
  Options:      json.RawMessage(`{"out": "stdout"}`),
  FormatOptions: json.RawMessage(`{"enable_color": true, "enable_caller": true}`),
  MaxQueueSize: 1000,
}
```
 And then call `logger.ConfigureTargets` passing those targets, and that should be it!

For file option, we need to configure log rotation properly with some reasonable defaults:

```
opts := struct {
		Filename    string `json:"filename"`
		Max_size    int    `json:"max_size"`
		Max_age     int    `json:"max_age"`
		Max_backups int    `json:"max_backups"`
		Compress    bool   `json:"compress"`
	}{
		Filename:    filename,
		Max_size:    100,
		Max_age:     0,
		Max_backups: 0,
		Compress:    true,
	}
json.Marshal(opts)
```